docs/source/kedro_project_setup/code_beyond_starter_files.md

```diff
@@ -0,0 +1,133 @@
+|-------|--------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1     | An ML model, or more precisely, a workflow to prepare the data, train an estimator, ship it to some registry | <ul> <li>Here Kedro fits well, as it allows to develop those pipelines in a modular and extensible way.</li> </ul>                                                                                                                                                                                                                                                                                               |
+| 2     | An optimiser that leverages the ML model and implements domain business logic to derive recommendations      | <ul> <li>A good design consideration might be to make it independent of the UI framework.</li> </ul>                                                                                                                                                                                                                                                                                                             |
+| 3     | User interface (UI) application                                                                              | <ul> <li>This can be a [`plotly`](https://plotly.com/python/) or [`streamlit`](https://streamlit.io/) dashboard.</li> <li>Or even a full-fledged front-end app leveraging JS framework like [`React`](https://react.dev/).</li> <li>Regardless, this component may know how to access the ML model, but it should probably not know anything about how it was trained and was Kedro involved or not.</li> </ul>  |
+
+A suggested solution in this case would be a **monorepo** design. Below is an example:
+
+```text
+└── repo_root
+    ├── packages
+    │   ├── kedro_project               <-- A Kedro project for ML model training.
+    │   │   ├── conf
+    │   │   ├── data
+    │   │   ├── notebooks
+    │   │   ├── ...
+    │   ├── optimizer                   <-- Standalone package.
+    │   └── dashboard                   <-- Standalone package, may import `optimizer`, but should not know anything about model training pipeline.
+    ├── requirements.txt                <-- Linters, code formatters... Not dependencies of packages.
+    ├── pyproject.toml                  <-- Settings for those, like `[tool.isort]`.
```

Comment:
Notice that having this astray `pyproject.toml` file here automatically communicates to `pip` that this is potentially a Python package. This has been an ongoing source of pain in the ecosystem, see my summary here https://github.com/kedro-org/kedro/issues/2280#issuecomment-1437071618

Now what's the best practice to manage tool configs in monorepos for Python given this issue... I have no idea, to be honest

Reply:
I removed `pyproject.toml` and rearranged it to this:

```text
└── repo_root
    ├── packages
    │   ├── kedro_project          <-- A Kedro project for ML model training.
    │   │   ├── conf
    │   │   ├── data
    │   │   ├── notebooks
    │   │   ├── ...
    │   ├── optimizer              <-- Standalone package.
    │   └── dashboard              <-- Standalone package, may import `optimizer`, but should not know anything about model training pipeline.
    ├── ...
    ├── ... Examples of what you may want in the repo root:
    ├── requirements.txt           <-- Linters, code formatters... Not dependencies of packages.
    ├── ruff.toml                  <-- Or configs for other tools that you want to share between packages.
    └── ...
```